src/sidebar/components/hypothesis-app.js

```diff
@@ -88,6 +90,20 @@ function HypothesisApp({
     }
   }, [isSidebar, profile, openSidebarPanel, settings]);
 
+  // Show a deprecation warning if current browser is IE11
+  useEffect(() => {
+    if (isIE11()) {
+      toastMessenger.notice(
+        'Hypothesis is ending support for this browser (Internet Explorer 11) on July 1, 2020.',
+        {
+          autoDismiss: false,
+          moreInfoURL:
+            'https://web.hypothes.is/help/which-browsers-are-supported-by-hypothesis/',
+        }
+      );
+    }
+  }, [toastMessenger]);
```

Comment:
Since this effect doesn't have any ties to the rest of the component, we could potentially extract it into an init function that is triggered from `src/sidebar/index.js` in future. Same for the existing effect above in fact. I think this is OK here for the moment though.